// @tatumio/tatum/dist/src/service/rpc/other/BnbLoadBalancerRpc.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var BnbLoadBalancerRpc_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.BnbLoadBalancerRpc = void 0;
/* eslint-disable @typescript-eslint/no-explicit-any */
const typedi_1 = require("typedi");
const generic_1 = require("../generic");
const AbstractBnbRpc_1 = require("./AbstractBnbRpc");
let BnbLoadBalancerRpc = BnbLoadBalancerRpc_1 = class BnbLoadBalancerRpc extends AbstractBnbRpc_1.AbstractBnbRpc {
    constructor(id) {
        super();
        this.loadBalancer = typedi_1.Container.of(id).get(generic_1.LoadBalancer);
    }
    destroy() {
        this.loadBalancer.destroy();
    }
    getRpcNodeUrl() {
        return this.loadBalancer.getActiveNormalUrlWithFallback().url;
    }
    rawRpcCall(body) {
        return this.loadBalancer.post({ body, path: '' });
    }
    post(post) {
        return this.loadBalancer.post(post);
    }
};
BnbLoadBalancerRpc = BnbLoadBalancerRpc_1 = __decorate([
    (0, typedi_1.Service)({
        factory: (data) => {
            return new BnbLoadBalancerRpc_1(data.id);
        },
        transient: true,
    }),
    __metadata("design:paramtypes", [String])
], BnbLoadBalancerRpc);
exports.BnbLoadBalancerRpc = BnbLoadBalancerRpc;
